Source/Dafny/Compiler-Csharp.cs

```diff
@@ -760,26 +842,23 @@ protected class ClassWriter : IClassWriter
       var customReceiver = createBody && !forBodyInheritance && NeedsCustomReceiver(m);
 
       AddTestCheckerIfNeeded(m.Name, m, wr);
-      wr.Write("{0}{1}{2}{3} {4}",
+      wr.Write("{0}{1}{2}{3} {4}{5}(",
         createBody ? "public " : "",
         m.IsStatic || customReceiver ? "static " : "",
         hasDllImportAttribute ? "extern " : "",
         targetReturnTypeReplacement ?? "void",
-        IdName(m));
-      if (m.TypeArgs.Count != 0) {
-        wr.Write("<{0}>", TypeParameters(m.TypeArgs));
-      }
-      wr.Write("(");
-      int nIns;
+        IdName(m),
+        TypeParameters(TypeArgumentInstantiation.ToFormals(ForTypeParameters(typeArgs, m, lookasideBody))));
+      var sep = "";
+      var nIns = WriteRuntimeTypeDescriptorsFormals(m, ForTypeDescriptors(typeArgs, m, lookasideBody), wr, ref sep, tp => $"{DafnyTypeDescriptor}<{tp.CompileName}> {FormatTypeDescriptorVariable(tp)}");
```

Comment:
Is there a reason we Write...Formals bout Emit...Actuals (and Emit most other things)?

Reply:
I think the `Emit...` methods started off as ones that were overridden by the individual compilers, whereas `Write...` methods were (non-overridable) worker routines. I'd be surprised if this "rule" is applied consistently. I will leave this unchanged for now, hoping for a future refactoring pass to clean up and apply consistency.